shared/src/main/java/uk/gov/di/authentication/shared/helpers/ApiGatewayResponseHelper.java

```diff
@@ -66,32 +78,40 @@ public static APIGatewayProxyResponseEvent generateEmptySuccessApiGatewayRespons
 
     public static APIGatewayProxyResponseEvent generateApiGatewayProxyResponse(
             int statusCode, String body, Map<String, List<String>> multiValueHeaders) {
-        APIGatewayProxyResponseEvent apiGatewayProxyResponseEvent =
-                new APIGatewayProxyResponseEvent();
-        apiGatewayProxyResponseEvent.setStatusCode(statusCode);
-        apiGatewayProxyResponseEvent.setBody(body);
+
+        var response =
+                new APIGatewayProxyResponseEvent()
+                        .withStatusCode(statusCode)
+                        .withBody(body)
+                        .withHeaders(SecurityHeaders.headers());
+
+        if (multiValueHeaders != null) {
+            response.setMultiValueHeaders(multiValueHeaders);
+        }
+
+        return response;
+    }
+
+    public static APIGatewayProxyResponseEvent generateApiGatewayProxyResponse(
+            int statusCode,
+            String body,
+            Map<String, String> headers,
+            Map<String, List<String>> multiValueHeaders) {
+
+        var allHeaders = SecurityHeaders.headers();
+
+        Optional.ofNullable(headers).ifPresent(allHeaders::putAll);
```

Comment:
Does this only add the security headers if there are other headers?  Presumably you always want the security headers even if there are no other headers?

Reply:
If there are other headers, then it adds them to the existing `allHeaders` variable, so it will always have those base headers.